core/lib/default_da_clients/src/object_store/client.rs

```diff
@@ -0,0 +1,95 @@
+use std::{
+    fmt,
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use zksync_config::ObjectStoreConfig;
+use zksync_da_client::{
+    types::{DAError, DispatchResponse, InclusionData},
+    DataAvailabilityClient,
+};
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::{pubdata_da::StorablePubdata, L1BatchNumber};
+
+/// An implementation of the `DataAvailabilityClient` trait that stores the pubdata in the GCS.
+#[derive(Clone)]
+pub struct ObjectStoreDAClient {
+    object_store: Arc<dyn ObjectStore>,
+}
+
+impl ObjectStoreDAClient {
+    pub async fn new(object_store_conf: ObjectStoreConfig) -> anyhow::Result<Self> {
+        Ok(ObjectStoreDAClient {
+            object_store: ObjectStoreFactory::create_from_config(&object_store_conf).await?,
+        })
+    }
+}
+
+#[async_trait]
+impl DataAvailabilityClient for ObjectStoreDAClient {
+    async fn dispatch_blob(
+        &self,
+        batch_number: u32,
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DAError> {
+        if let Err(err) = self
+            .object_store
+            .put(L1BatchNumber(batch_number), &StorablePubdata { data })
+            .await
+        {
+            return Err(DAError {
+                is_transient: err.is_transient(),
+                error: anyhow::Error::from(err),
+            });
+        }
+
+        Ok(DispatchResponse {
+            blob_id: batch_number.to_string(),
+        })
+    }
+
+    async fn get_inclusion_data(&self, key: String) -> Result<Option<InclusionData>, DAError> {
+        let key_u32 = key.parse::<u32>().map_err(|err| DAError {
+            error: anyhow::Error::from(err),
```

Comment:
Nit: In this case, using `anyhow::Context::context()` extension could allow providing more error context, i.e. that an error has occurred specifically when parsing a `key` (could be included in the context) as an L1 batch number.

Reply:
done